platform/saivpp/vpplib/SwitchStateBaseBfd.cpp

```diff
@@ -0,0 +1,367 @@
+    _In_ sai_object_id_t switch_id,
+    _In_ uint32_t attr_count,
+    _In_ const sai_attribute_t *attr_list)
+{
+    int ret = 0;
+    SWSS_LOG_ENTER();
+
+    sai_object_id_t bfd_oid;
+    sai_deserialize_object_id(serializedObjectId, bfd_oid);
+
+    sai_attribute_t attr;
+    /* Attribute#1 */
+    attr.id = SAI_BFD_SESSION_ATTR_MIN_RX;
+    CHECK_STATUS(get(SAI_OBJECT_TYPE_BFD_SESSION, bfd_oid, 1, &attr));
+    uint32_t required_min_rx = attr.value.u32;
+
+    /* Attribute#2 */
+    attr.id = SAI_BFD_SESSION_ATTR_MIN_TX;
+    CHECK_STATUS(get(SAI_OBJECT_TYPE_BFD_SESSION, bfd_oid, 1, &attr));
+    uint32_t required_min_tx = attr.value.u32;
+
+    /* Attribute#3 */
+    attr.id = SAI_BFD_SESSION_ATTR_MULTIPLIER;
+    CHECK_STATUS(get(SAI_OBJECT_TYPE_BFD_SESSION, bfd_oid, 1, &attr));
+    uint8_t detect_mult = attr.value.u8;
+
+    /* Attribute#4 */
+    attr.id = SAI_BFD_SESSION_ATTR_SRC_IP_ADDRESS;
+    CHECK_STATUS(get(SAI_OBJECT_TYPE_BFD_SESSION, bfd_oid, 1, &attr));
+    sai_ip_address_t local_addr = attr.value.ipaddr;
+    vpp_ip_addr_t vpp_local_addr;
+    /*local addr */
+    sai_to_vpp_ip_addr(&vpp_local_addr, &local_addr);
+
+    /* Attribute#5 */
+    attr.id = SAI_BFD_SESSION_ATTR_DST_IP_ADDRESS;
+    CHECK_STATUS(get(SAI_OBJECT_TYPE_BFD_SESSION, bfd_oid, 1, &attr));
+    sai_ip_address_t peer_addr = attr.value.ipaddr;
+    vpp_ip_addr_t vpp_peer_addr;
+    /* Peer Addr*/
+    sai_to_vpp_ip_addr(&vpp_peer_addr, &peer_addr);
+
+    /* Attribute#6 */
+    attr.id = SAI_BFD_SESSION_ATTR_MULTIHOP;
+    CHECK_STATUS(get(SAI_OBJECT_TYPE_BFD_SESSION, bfd_oid, 1, &attr));
+    bool multihop = attr.value.booldata;
+    
+    const char *hwif_name = NULL;
+    if (!multihop) {
+        /* Attribute#7 */
+        attr.id = SAI_BFD_SESSION_ATTR_PORT;
+        CHECK_STATUS(get(SAI_OBJECT_TYPE_BFD_SESSION, bfd_oid, 1, &attr));
+        sai_object_id_t port_id = attr.value.oid;
+        sai_object_type_t obj_type = objectTypeQuery(port_id);
+        if (obj_type != SAI_OBJECT_TYPE_PORT)
+        {
+            SWSS_LOG_ERROR("SAI_BRIDGE_PORT_ATTR_PORT_ID=%s expected to be PORT but is: %s",
+                    sai_serialize_object_id(port_id).c_str(),
+                    sai_serialize_object_type(obj_type).c_str());
+            return SAI_STATUS_FAILURE;
+        }
+        std::string ifname = "";
+        if (vpp_get_hwif_name(port_id, 0, ifname) == true)
+        {
+            hwif_name = ifname.c_str();
+        }
+        else
+        {
+            SWSS_LOG_ERROR("BFD session create request FAILED due to invalid hwif name");
+    
+            return SAI_STATUS_FAILURE;
+        }
+    }
+
+    if (multihop || hwif_name) {
+        SWSS_LOG_NOTICE("BFD session create request sent to VPP, hwif: %s, multihop: %d, oid: %lld",  hwif_name, multihop, bfd_oid);
+        /*vpp call to add bfd session*/
+        ret = bfd_udp_add(multihop, hwif_name, &vpp_local_addr, &vpp_peer_addr, detect_mult, required_min_tx, required_min_rx);
```

Comment:
If you care about return, you need to change W(ret) in bfd_udp_add to WR(ret). W(ret) just returns the last result in the queue, which may not be the reply of your bfd request.